### DISCUSSION TITLE
Error in Quarto Document RStudio

### DISCUSSION THREAD

**Discussion by andrewanggoh** [teamalgoritma/community#924] [Q&A] [answered]
Dear Team Algoritma,

Saya mengalami kendala/error ketika menggunakan Python dalam Quarto Document File di RStudio. Error yang muncul adalah sebagai berikut :

![image](https://github.com/user-attachments/assets/d4b1e3be-8845-4194-ac81-6225ec9a79ce)

Selagi code tersebut dalam proses running muncul pop up untuk melakukan "Create a default Python environment..." seperti berikut :

![image](https://github.com/user-attachments/assets/ddc5a649-1d93-47af-b76a-4d7b8a7bc7d5)

Saya udah memilih "Yes" namun tidak ada perubahan dan tetap error seperti gambar sebelumnya. Untuk Python Interpreter di Global Options sudah saya sesuaikan.

![image](https://github.com/user-attachments/assets/e77a2780-35aa-45db-94c7-f508766f7238)

Mohon bantuannya. Terima kasih banyak sebelumnya 🙏🏼

Best regards,
Andrew

**Comment by VictorNugraha** [comment]
halo Pak @andrewanggoh, 

Untuk kendala yang terjadi sepertinya dikarenakan koneksi antara RStudio dengan Python/Anaconda.

Satu hal yang ingin saya coba konfirmasi adalah, apakah Bapak sudah melakukan tahapan seperti yang terlampir di bawah?
![image](https://github.com/user-attachments/assets/74528fa9-7c38-4473-92fa-802063232bed)

Mohon konfirmasinya untuk tahapan di atas, terima kasih

**Reply by andrewanggoh** [reply]
Sudah Pak, sudah saya lakukan tahapan tersebut

**Reply by VictorNugraha** [reply]
untuk kendalanya apakah masih sama walaupun sudah melakukan tahapan tersebut?

**Reply by andrewanggoh** [reply]
Iya betul Pak, masih sama

**Reply by IrfanChairurrachman** [reply, marked as answer]
dear @andrewanggoh ,

Hal ini mungkin terjadi karena R Studio tidak mengenali conda environment pada global options. Salah satu cara yg bisa kita lakukan adalah konfigurasi menggunakan `Sys.env` di R Studio Console, cara yang dapat dilakukan:

1. Jalankan command berikut pada R Studio Console: `reticulate::conda_list()`

Output yang dihasilkan adalah list environment python yang anda miliki, seperti di bawah ini:
![image](https://github.com/user-attachments/assets/13f9cc64-7820-4573-bad8-9da6640cb9a2)

2. Pilih salah satu path / environment yang ingin Anda gunakan, misalnya dalam hal ini saya memilih env `xion_da`, maka copy path xion_da kemudian jalankan command berikut: `Sys.setenv(RETICULATE_PYTHON = "C:\\Users\\Lenovo\\anaconda3\\envs\\xion_da/python.exe")`

Setelah itu kemudian dijalankan lagi python di R Studionya.

Sekian jawaban dari kami, semoga membantu

**Reply by IrfanChairurrachman** [reply]
Dear @andrewanggoh 

Untuk menjaga agar forum diskusi berjalan kondusif, karena tidak ada respon lanjutan setelah 7 hari dari respon terakhir, kami izin menandai pertanyaan sudah terjawab dan menutup diskusi ini. Semoga dapat dipahami dan terima kasih.

Best regards,
Team Algoritma
